File: packages/test-equivalence-matchers/src/boundaries/nodes/modifiers/ShapedLike.ts
import { AnyMatcherOrData } from '@contract-case/case-entities-internal';
import { CascadingContextMatcher } from '../base';

/**
 * Everything inside this matcher will be matched on the shape of the data (ie,
 * type alone), unless overridden with other matchers.  Use this to switch out
 * of the default `exactlyLike` matching.
 */
export class ShapedLike extends CascadingContextMatcher {
  /**
   * @param content - The object, array, primitive or matcher to match the shape against
   */
  constructor(content: AnyMatcherOrData) {
    super(content, { matchBy: 'type' }, {});
  }

  /**
   * For non-TypeScript implementations (see [AnyMatcher.toJSON()](#\@case-contract-testing/test-equivalence-matchers.AnyMatcher.toJSON))
   *
   * @privateRemarks
   * This comment and the implementation is boilerplate on all matchers to avoid
   * outputting duplicate unimportant documentation on all matcher classes of
   * the docs. Only modify this comment or the implementation via search and replace.
   */
  override toJSON(): unknown {
    return super.toJSON();
  }
}

File: packages/dsl-java/src/main/java/io/contract_testing/contractcase/dsl/matchers/modifiers/ShapedLike.java
package io.contract_testing.contractcase.dsl.matchers.modifiers;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.contract_testing.contractcase.dsl.ContractCaseDsl;
import io.contract_testing.contractcase.dsl.DslMatcher;
import jakarta.annotation.Generated;
import java.lang.Object;
import java.lang.String;
import lombok.Builder;
import lombok.Getter;
import lombok.Getter;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;

/**
 * Everything inside this matcher will be matched on the shape of the data (ie,
 * type alone), unless overridden with other matchers. Use this to switch out of the default {@code exactlyLike} matching.
 */
@Generated("@contract-case/case-definition-generator")
@ContractCaseDsl
public class ShapedLike implements DslMatcher {

  /**
   * ContractCase's internal type for this element
   */
  @Getter
  @JsonProperty("_case:matcher:type")
  private final String type;

  /**
   * The object, array, primitive or matcher to match the shape against
   */
  @Getter
  @JsonProperty("_case:matcher:child")
  private final Object child;

  @Getter
  @JsonProperty("_case:context:matchBy")
  @JsonInclude(Include.ALWAYS)
  public final String matchBy = "type";

  @Builder
  public ShapedLike(@NotNull final Object child) {
    this.type = "_case:CascadingContext";
    this.child = child;
  }
}
